code/mob/living/critter/adventure.dm

```diff
@@ -77,6 +77,7 @@
 			if (isintangible(C)) continue
 			if (isdead(C)) continue
 			if (istype(C, src.type)) continue
+			if (C.faction & src.faction != 0 && (C.faction == src.faction)) continue //Checks if they are in the same faction
```

Comment:
Not sure what you think this is doing, but one half of it is definitely redundant. Also somewhat out of scope, but this practice of copy pasting a bunch of checks into `seek_target` for every new mobcritter is bad and should probably be a `valid_target` proc instead.

Reply:
Did this as suggested by amylizzle in a previous comment. I believe it prevents factionless mobs from being friendly to other factionless mobs. I could also remove the copy-pasted code from various critters and save that for a future pr, but that will mean that you cannot implement a faction for these mobs until that is pr-ed.

Reply:
And I mean save changing the copy-pasted code into a proc for a future pr, not just copy-pasting.

Reply:
Right, but why are you doing the bitwise check and then checking exact equality as well? I think Amy's suggestion was to do `if (src.faction != null && (C.faction & src.faction != 0))` since that would cause null factions to always be aggressive, and any mobs that share factions to be passive.

Reply:
If I only had it check ```if (src.faction != null && (C.faction & src.faction != 0))```, then that would mean that a SYNDICATE faction mob and a NANOTRASEN faction mob would ignore each other, cause they both meet this if check, even if they would not be on the same team, unless I'm misunderstanding how bitmask checks work.

Reply:
You are misunderstanding how bit checks work. `1 & 2 = 0`, but `3 & 2 = 2` so syndie and NT would be enemies, but a mob could be both syndie and NT and be friendly to both.

Reply:
Ok, I believe I've changed them to what ya'll suggested